doc/source/tutorial/stats/sampling.rst

```diff
@@ -0,0 +1,202 @@
+
+
+Basic concepts of the Interface
+-------------------------------
+
+Every generator needs to be set up before one can start sampling from it.
+This can be done by instantiating an object of that class. Most of the
+generators take a distribution object as input which contains the implementation
+of required methods like PDF, CDF, etc. In addition to the distribution
+object, one can also pass parameters used to set up the generator. It is also
+possible to truncate the distributions using a ``domain`` parameter.  All
+generators need a stream of uniform random numbers that are transformed into
+random variates of the given distribution. This is done by passing a ``seed``
+parameter with a NumPy BitGenerator as the uniform random number generator.
+``seed`` can either be a integer, `np.random.Generator`,
+`np.random.BitGenerator`, `np.random.RandomState`, or
+`np.random.SeedSequence`.
+
+.. warning:: Use of NumPy < 1.19.0 is discouraged as it doesn't have a fast
+             Cython API for generating uniform random numbers and might be
+             too slow for practical use.
+
+All the generators have a common ``rvs`` method that can be used to draw
+samples from the given distribution.
+
+An example of this interface is shown below:
+
+    >>> from scipy.stats import TransformedDensityRejection
+    >>> from math import exp
+    >>> 
+    >>> class StandardNormal:
+    ...     def pdf(self, x: float) -> float:
+    ...         # note that the normalization constant isn't required
+    ...         return exp(-0.5 * x*x)
+    ...     def dpdf(self, x: float) -> float:
+    ...         return -x * exp(-0.5 * x*x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> 
+    >>> urng = np.random.default_rng(0x6be00a9336fc82258fe73bfa672d3d52)
+    >>> rng = TransformedDensityRejection(dist, seed=urng)
+
+As shown in the example, we first initialize a distribution object that
+contains an implementation of the methods required by the generator. In
+our case, we use the :class:`~TransformedDensityRejection` method which
+requires a PDF and its derivative w.r.t. x (i.e. the variate).
+
+.. note:: Note that the methods of the distribution (i.e. ``pdf``,
+          ``dpdf``, etc) need not be vectorized. They should
+          accept and return floats.
+
+.. note:: One can also pass the SciPy distributions as arguments but it can
+          be slow due to validations and expensive NumPy operations.
+          Moreover, it doesn't always have all the information required
+          by some generators like derivative of PDF for TDR method. Also,
+          most of the distributions in SciPy provide a ``rvs`` method which
+          can be used instead.
+
+In the above example, we have set up an object of the
+:class:`~TransformedDensityRejection` method to sample from a
+standard normal distribution. Now, we can start sampling from our
+distribution by calling the ``rvs`` method:
+
+    >>> rng.rvs()
+    -0.6853213192313732
+    >>> rng.rvs((5, 3))
```

Comment:
> Instead of this, I would show some plots instead.

That's a nice idea. This is missing pretty plots. Adding those should make this more interesting.

Reply:
To get you started I have this for instance

```python
fig, ax = plt.subplots()
x = np.linspace(distribution.ppf(0.01), distribution.ppf(0.99), 100)
# PDF
pdf = distribution.pdf(x)
ax.plot(x, pdf, "-", lw=5, alpha=0.6)
# Empirical PDF
sample = distribution.rvs(size=1000)
# discrete samples
delta = np.max(pdf) * 5e-2
ax.plot(sample[:100], -delta - delta * np.random.random(100), "+k")
ax.set_title("PDF")
ax.set_ylabel(r"$f$")
ax.set_xlabel(r"$x$")
```